In [1]:
1+1

2

In [5]:
list(range(10))

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

In [6]:
list(range(2,22))

[2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21]

In [7]:
name='yabuki'

In [8]:
len(name)

6

In [9]:
name.upper()

'YABUKI'

In [10]:
len([0,1,2,3,4])

5

len(name)とname.upper()に統一感はない

len()はほかのことにも使える
len([0,1,2,3,4])
upper()は文字列用

In [11]:
address = 'Tokyo,Japan'
address.split(',')

['Tokyo', 'Japan']

In [12]:
address.split('o')

['T', 'ky', ',Japan']

split : 指定された文字列で分割する

In [13]:
address.index('T')

0

In [14]:
address.index('X')

ValueError: substring not found

indexは検索　何文字目にあるか分かる

In [15]:
address.index('o')

1

文字列は０から数える。Oは2つあるが左から探すので１としか出ない

In [16]:
address.upper()

'TOKYO,JAPAN'

メソットは全て()を付ける　upper:大文字にする

In [19]:
address = 'FCBARCELONA'
address.lower()

'fcbarcelona'

lower：小文字にする

ここから時刻を扱う

In [20]:
from datetime import *
date_a = date(2008, 1, 1)
date_b = date(2008, 1, 5)
result = date_b - date_a
print(result) 

4 days, 0:00:00


引き算をやっている
答えは何日後かが分かる

In [22]:
from datetime import *
date_a = date(2008, 1, 1)
date_b = date(2009, 1, 1)
result = date_b - date_a
print(result) 

366 days, 0:00:00


In [23]:
date_a.year

2008

date_a 366日それは何年ですか？

name.upper()
属性あり：何らかの処理をする
↑メソッド
date_a.year
括弧なし：データにアクセスするだけ
↑属性

In [24]:
date_a.month

1

In [25]:
date_a.day

1

In [27]:
date.today()

datetime.date(2017, 5, 17)

In [32]:
birthday = date(1997,5,30)

In [34]:
today = date.today()

In [35]:
result = today - birthday

In [36]:
print(result)

7292 days, 0:00:00


In [40]:
today.weekday()

2

In [41]:
date_a

datetime.date(2008, 1, 1)

In [43]:
date_a.weekday()

1

大きな数をやる前に小さい確認できる数で動作が正しいか確かめる

１が火曜日

In [45]:
date_c = date(2000,5,5)

In [46]:
tmp = '2000,5,5'

In [47]:
tmp2 = [2000,5,5]

45.46.47は全て違う　パイソンとして使いやすいのは45のみ

おわり